app/models/agents/naive_bayes_agent.rb

```diff
@@ -0,0 +1,509 @@
+require 'yaml'
+
+# NBayes from Oasic Technologies
+# https://github.com/oasic/nbayes
+#
+# Robust implementation of NaiveBayes:
+# - using log probabilities to avoid floating point issues
+# - Laplacian smoothing for unseen tokens
+# - allows binarized or standard NB
+# - allows Prior distribution on category to be assumed uniform (optional)
+# - generic to work with all types of tokens, not just text
+
+
+module NBayes
```

Comment:
I'd recommend adding rspec coverage for your agent too, whether or not it's added to Huginn directly or pulled into it's own gem.